scripts/cleanup/unusued_compilation.py

```diff
@@ -0,0 +1,136 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import argparse
+import fnmatch
+import os
+import shutil
+import sys
+sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../build'))
+import ci_build
+
+EXTENSIONS_OF_INTEREST = ('.c', '.cc', '.cpp', '.cxx', '.h', '.hpp', '.hxx', '.inl')
+EXCLUSIONS = (
+    'AZ_DECLARE_MODULE_CLASS(', 
+    'REGISTER_QT_CLASS_DESC(',
+    'TEST(',
+    'TEST_F(',
+    'TEST_P(',
+    'INSTANTIATE_TEST_CASE_P(',
+    'INSTANTIATE_TYPED_TEST_CASE_P(',
+    'AZ_UNIT_TEST_HOOK(',
+    'IMPLEMENT_TEST_EXECUTABLE_MAIN(',
+    'BENCHMARK_REGISTER_F(',
+    'DllMain(',
+    'wWinMain(',
+    'AZ_DLL_EXPORT',
+    'CreatePluginInstance('
+)
+PATH_EXCLUSIONS = (
+    '*\\Platform\\Android\\*',
+    '*\\Platform\\Common\\*',
+    '*\\Platform\\iOS\\*',
+    '*\\Platform\\Linux\\*',
+    '*\\Platform\\Mac\\*',
+    'Templates\\*',
+    'python\\*',
+    'build\\*',
+    'install\\*',
+    'Code\\Framework\\AzCore\\AzCore\\Android\\*',
+    'Gems\\ImGui\\External\\ImGui\\*',
+    '*_Traits_*.h',
+)
+
+
+def create_filelist(path):
+    filelist = set()
+    for input_file in path:
+        if os.path.isdir(input_file):
+            for dp, dn, filenames in os.walk(input_file):
+                if 'build\\windows_vs2019' in dp:
+                    continue
+                for f in filenames:
+                    extension = os.path.splitext(f)[1]
+                    extension_lower = extension.lower()
+                    if extension_lower in EXTENSIONS_OF_INTEREST:
+                        normalized_file = os.path.normpath(os.path.join(dp, f))
+                        filelist.add(normalized_file)
+        else:
+            extension = os.path.splitext(input_file)[1]
+            extension_lower = extension.lower()
+            if extension_lower in EXTENSIONS_OF_INTEREST:
+                normalized_file = os.path.normpath(os.path.join(os.getcwd(), input_file))
+                filelist.add(normalized_file)
+            else:
+                print(f'Error, file {input_file} does not have an extension of interest')
```

Comment:
minor: 
If its possible to have a large number of files, maybe report all of the bad files first and then exit from this function